### PR TITLE
Adds Azure integration

### PR DESCRIPTION
**Pull request by noelbundick**
Integrates laboratory + Azure bits so that it can be deployed to Azure App Service. Still runs in-proc by default.

Other updates
* Improves VS Code test integration. Ex: 
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,57 +4,66 @@
...
-    "test-functional": "mocha build/test/functional --config test/functional/.mocharc.yml",
+    "test": "mocha",
+    "pretest:functional": "npm run compile",
+    "test:functional": "mocha --config test/functional/.mocharc.yml",
     "check": "gts check",
-    "clean": "gts clean",
+    "clean": "gts clean && rimraf ./dist",
     "compile": "tsc -p .",
+    "dct": "node build/src/cli/dct.js",
+    "dct:dev": "npm run compile && node -r dotenv/config build/src/cli/dct.js",
```
Comment (MikeHopcroft) [inline package.json:14]: Just curious - is the ":dev" a convention or something that has semantics for npm?

File: package.json [hunk 2]
```diff
@@ -4,57 +4,66 @@
...
     "fix": "gts fix",
     "prepare": "npm run compile",
     "pretest": "npm run compile",
     "posttest": "npm run check",
-    "worker": "node build/src/worker/app.js"
+    "laboratory": "node build/src/laboratory/server/main.js",
+    "laboratory:dev": "npm run compile && node -r dotenv/config build/src/laboratory/server/main.js",
+    "laboratory:package:appservice": "npm run compile && mkdir -p dist && cp -r --parents package*.json build/src dist && cd dist && npm install --ignore-scripts --production && zip -r $npm_package_name.zip package.json package-lock.json build/src node_modules",
+    "worker": "node build/src/worker/app.js",
+    "worker:dev": "npm run compile && node -r dotenv/config build/src/worker/app.js"
```
Comment (MikeHopcroft) [inline package.json:23]: So the idea here is that the codebase won't know about dotenv? Is the intention to use the .env file when doing dev work, and let Azure set the environment variables in the cloud?

File: scripts/laboratory.rest [hunk 3]
```diff
@@ -0,0 +1,101 @@
+# List benchmarks
```
Comment (MikeHopcroft) [inline scripts/laboratory.rest:1]: This is a nice tool!

File: package.json (hunk 1 above)
Comment (noelbundick) [reply package.json:14]: not particularly, it's just a delimiter I've seen before. To npm, it's an independent task and just an opaque string

File: package.json (hunk 2 above)
Comment (noelbundick) [reply package.json:23]: Exactly

**Pull request merged**